cmd/lotus-storage-miner/workers.go

```diff
@@ -0,0 +1,87 @@
+package main
+
+import (
+	"fmt"
+	"github.com/filecoin-project/lotus/api"
+	"github.com/filecoin-project/lotus/chain/types"
+	"gopkg.in/urfave/cli.v2"
+	"sort"
+
+	lcli "github.com/filecoin-project/lotus/cli"
+)
+
+var workersCmd = &cli.Command{
+	Name:  "workers",
+	Usage: "interact with workers",
+	Subcommands: []*cli.Command{
+		workersListCmd,
+	},
+}
+
+var workersListCmd = &cli.Command{
+	Name:  "list",
+	Usage: "list workers",
+	Action: func(cctx *cli.Context) error {
+		nodeApi, closer, err := lcli.GetStorageMinerAPI(cctx)
+		if err != nil {
+			return err
+		}
+		defer closer()
+
+		ctx := lcli.ReqContext(cctx)
+
+		stats, err := nodeApi.WorkerStats(ctx)
+		if err != nil {
+			return err
+		}
+
+		st := make([]struct {
+			id uint64
+			api.WorkerStats
+		}, 0, len(stats))
```

Comment:
You can just define this type within the function. This way you won't have to repeat type definition and it won't leak.